packages/sdk/src/AbstractConfigManager.ts

```diff
@@ -0,0 +1,654 @@
+
+        await this.withProgress("Validating Private Keys...", async (progress) => {
+            // Find private keys located at ~/.ssh/ and attempt to connect with them
+            if (!this.validationResult) {
+                const foundPrivateKeys = await ZClientUtils.findPrivateKeys();
+                for (const privateKey of foundPrivateKeys) {
+                    const testValidation: ISshConfigExt = this.selectedProfile!;
+                    testValidation.privateKey = privateKey;
+                    const result = await this.validateConfig(testValidation);
+                    progress(100 / foundPrivateKeys.length);
+
+                    if (result) {
+                        this.validationResult = {};
+                        if (Object.keys(result).length >= 1) {
+                            this.selectedProfile = { ...this.selectedProfile, ...result };
+                        }
+                        return;
+                    }
+                }
+            }
+
+            // Match hostname to configurations from ~/.ssh/config file
+            let validationAttempts = this.migratedConfigs.filter(
+                (config) => config.hostname === this.selectedProfile?.hostname,
+            );
+
+            // If multiple matches exist, narrow down by user
+            if (validationAttempts.length > 1 && this.selectedProfile?.user) {
+                validationAttempts = validationAttempts.filter((config) => config.user === this.selectedProfile?.user);
+            } else {
+                // If no user is specified, allow all configs where the hostname matches
+                validationAttempts = validationAttempts.filter(
+                    (config) => !this.selectedProfile?.user || config.user === this.selectedProfile?.user,
+                );
+            }
+
+            for (const profile of validationAttempts) {
+                const testValidation: ISshConfigExt = profile;
+                const result = await this.validateConfig(testValidation);
+                progress(100 / validationAttempts.length);
+                if (result !== undefined) {
+                    this.validationResult = {};
+                    this.selectedProfile = {
+                        ...this.selectedProfile,
+                        privateKey: testValidation.privateKey,
+                    };
+                    if (Object.keys(result).length >= 1) {
+                        this.selectedProfile = {
+                            ...this.selectedProfile,
+                            ...result,
+                        };
+                    }
+                    return;
+                }
+            }
+        });
+    }
+
+    private async setProfile(selectedConfig: ISshConfigExt | undefined, updatedProfile?: string): Promise<void> {
+        const configApi = this.mProfilesCache.getTeamConfig().api;
+        // Create the base config object
+        const config = {
```

Comment:
If we strongly type this object, then the `@ts-ignore` comment below can be removed 😋 
```suggestion
        const config: IConfigProfile = {
```